### PR TITLE
電線接続範囲を相互判定方式へ刷新しEnergized Range表示を廃止

### PR DESCRIPTION
**Pull request by sakastudio**
# 📌 diff の参照は今後こちら → [`49f93cf338...0542f9645f`](https://github.com/moorestech/moorestech/compare/49f93cf338f98abb99c4dff06607573e3e90c064...0542f9645f0e26c8c52da4b8a4728ac38c7c486e)
### マージコミット → [`3e69bf093c`](https://github.com/moorestech/moorestech/commit/3e69bf093c05a9564624cc1ce6c879092f1c62ed)

---

## Summary
- 電線・電柱・機械の接続可否判定を、片側の最大距離のみで見る方式から「双方の接続範囲ボックスが重なるか」を見る相互判定方式（IsMutuallyConnectable）へ刷新
- 廃止予定だった接続範囲の可視化表示（DisplayEnergizedRange・EnergizedRangeObject関連）とMaxElectricPoleMachineConnectionRange・maxWireLengthを設計から全廃し、プレビュー線表示に一本化
- 電線関連の単体・結合・E2E（設置時自動接続シナリオ、証拠動画付き）テストを新方式に合わせて追加・再設計

## Test plan
- [x] `uloop compile --project-path ./moorestech_client` でコンパイル確認
- [x] `uloop run-tests --project-path ./moorestech_client --filter-type regex --filter-value "ElectricWire|ElectricConnectionRangeService"` でサーバー側テスト確認
- [x] 電線相互範囲判定の設置時自動接続シナリオをUnity PlayMode E2Eで実行し証拠動画・スクリーンショットを記録

Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Electric wires now auto-connect only when both endpoints are mutually in range.
  * Auto-connect selects the nearest eligible pole first, then eligible unconnected machines (including clean-room machines).
* **Bug Fixes**
  * Improved electric-wire placement/extend validation with mutual horizontal/vertical range and correct multi-block overlap.
  * Wired placement UI no longer displays energized-range indicators, and the web placement payload no longer includes that visibility flag.
  * Out-of-range failures now report as “OutOfRange” (instead of “TooFar”).
* **Tests**
  * Added/updated coverage for mutual connectability, boundary cases, selector targeting, and revised failure reasons.

### CODE REVIEW

File: moorestech_client/Assets/Scripts/Client.Game/InGame/BlockSystem/PlaceSystem/Common/ElectricWireAutoConnect/ClientElectricWireAutoConnectCollector.cs [hunk 1]
```diff
@@ -16,42 +15,36 @@ namespace Client.Game.InGame.BlockSystem.PlaceSystem.Common.ElectricWireAutoConn
     /// <summary>
     /// 自動接続候補を受信済みクライアント状態のみで収集する。サーバーのワールド状態には触れない
     /// Collects auto-connect candidates using only received client state; never touches server world state
-    /// 収集ルールはサーバーのElectricWireAutoConnectTargetCollectorと同一（最寄り電柱1本＋未接続機械）
-    /// The selection rules mirror the server's ElectricWireAutoConnectTargetCollector (nearest pole + unconnected machines)
-    /// 注意: クライアント辞書は原点座標登録のみのため、複数セルブロックの範囲判定は原点基準の近似となる
-    /// Note: the client dictionary registers origin cells only, so range checks for multi-cell blocks approximate by origin
+    /// 収集ルールはサーバーのElectricWireAutoConnectTargetCollectorと同一（相互範囲判定＋最寄り電柱1本＋未接続機械）
+    /// The rules mirror the server's ElectricWireAutoConnectTargetCollector (mutual range, nearest pole, unconnected machines)
```
Comment (sakastudio) [inline moorestech_client/Assets/Scripts/Client.Game/InGame/BlockSystem/PlaceSystem/Common/ElectricWireAutoConnect/ClientElectricWireAutoConnectCollector.cs]: これってそもそもサーバー側とロジックを共通化できない？
二重ロジックになってる

File: VanillaSchema/blocks.yml [hunk 2]
```diff
@@ -286,9 +286,12 @@ properties:
...
           default: 2
-        - key: maxWireLength
-          type: number
-          default: 8
+        - key: connectionRange
+          type: integer
+          default: 30
+        - key: connectionHeightRange
+          type: integer
+          default: 20
```
Comment (sakastudio) [inline VanillaSchema/blocks.yml:306]: これ、そもそもinterfaceに組み込めないかな

**Review by sakastudio — COMMENTED** [review]
要チェック
